### PR TITLE
Improve man pages

### PR DESCRIPTION
**Pull request by eepp**

### CODE REVIEW

File: doc/man/lttng-ust.3.txt [hunk 1]
```diff
@@ -563,23 +563,52 @@ named 't_name' from the provider named 'prov_name' is enabled at
...
+Here's how to use `tracepoint_enabled()` and `do_tracepoint()`:
+
+------------------------------------------------------------------------
+if (tracepoint_enabled(my_provider, my_tracepoint)) {
+    payload = prepare_payload();
+    do_tracepoint(my_provider, my_tracepoint, payload);
+}
 ------------------------------------------------------------------------
+
+Using `tracepoint()` with `tracepoint_enabled()` is dangerous since
```
Comment (compudj) [inline doc/man/lttng-ust.3.txt]: No it's not dangerous.

File: doc/man/lttng-ust.3.txt [hunk 2]
```diff
@@ -563,23 +563,52 @@ named 't_name' from the provider named 'prov_name' is enabled at
...
+tracepoint(my_provider, my_tracepoint, payload);
 ------------------------------------------------------------------------
 
-If the tracepoint is enabled after the condition, then `stuff` is not
+If the tracepoint is enabled after the condition, then `payload` is not
 prepared: the emitted event will either contain wrong data, or the
 whole application could crash (segmentation fault, for example).
 
+If you use `tracepoint()` within the conditional block, then it's
+possible that `payload` is prepared for nothing:
```
Comment (compudj) [inline doc/man/lttng-ust.3.txt]: It's inefficient, but not dangerous.

What is dangerous is this:

if (tracepoint_enabled(...)) { setup stuff }
tracepoint(use stuff)

Comment (eepp) [reply doc/man/lttng-ust.3.txt]: That's exactly what the "dangerous" situation is about above. This example is indeed not dangerous, but it's not introduced as dangerous, it just says that the payload could be prepared for nothing (inefficient).

**Comment by mjeanson** [comment]
Closing for inactivity, if you feel this is still relevant, please reopen the PR.

**Pull request closed without merging**